webrpc.go

```diff
@@ -26,24 +27,45 @@ func ParseSchemaFile(path string) (*schema.WebRPCSchema, error) {
 		return schema.ParseSchemaJSON(json)
 
 	case ".ridl":
-		// Use root FS to allow RIDL file imports from parent directories,
-		// ie. import ../../common.ridl.
-
-		root := "/"
-
-		// Support Windows paths. Currently only supports paths on the same volume.
-		if volume := filepath.VolumeName(absolutePath); volume != "" {
-			root = volume + "/"
+		root, _, err := getRootPath()
+		if err != nil {
+			return nil, fmt.Errorf("get root path: %w", err)
 		}
 
-		path := filepath.ToSlash(absolutePath[len(root):])
+		schema, err := ridl.NewParser(os.DirFS(root), filepath.ToSlash(absolutePath[len(root):])).Parse()
+		if err != nil {
+			return nil, err
+		}
 
-		rootFS := os.DirFS(root)
+		// Convert absolute paths to relative paths.
+		if root, wd, _ := getRootPath(); wd != "" {
+			basePath := strings.TrimPrefix(wd, root)
+			schema.Filename, _ = filepath.Rel(basePath, schema.Filename)
+			for _, t := range schema.Types {
+				if filename, _ := filepath.Rel(basePath, t.Filename); filename != "" {
+					t.Filename = filename
+				}
+			}
+		}
```

Comment:
instead of post-processing the paths in schema object
(i don't think we need the filenames in the webrpc schema at all)

I think I'd prefer the RIDL parser to error out with absolute paths if it ever sees a duplicate type

Reply:
The duplicate detection error happens in the type parsing phase.
The error already shows the full path. This is the output of one of the tests:
```
=== RUN   TestRIDLDuplicateEnumSameFile
    ridl_test.go:218: RIDL duplicate enum same file:
        schema error: type 'Foo' was already declared at /schema/c.ridl:6 and /schema/c.ridl:9
        stack trace:
          - /schema/c.ridl
          - /schema/b.ridl
          - /schema/a.ridl
```
After the parsing has ben completed successfully, paths are transformed to relative to the WD, so that the debug output produces the same result with the same command, not depending on the path of the the repo itself.